### PR TITLE
REM-430

### PR DESCRIPTION
**Pull request by imtithalhalalo**
iv-measurements
Measurement Tab
Measurement Unit Tab

### CODE REVIEW

File: src/pages/iv-measurements/index.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          isLoading={false}
+          pageSize={50}
+          paginationType='api'
+          paginationParameters={paginationParameters}
+          refetch={refetch}
+          maxAccess={access}
+        />
+      </Grow>
+      {windowOpen && (
+        <MeasurementWindow
```
Comment (RebeccaFrem02) [inline src/pages/iv-measurements/index.js:102]: this is an old way of writing windows, now we use window stack

**Comment by FatimaSalami** [comment]
pr

i can't edit old records, but for new data i can

File: src/pages/iv-measurements/Windows/MeasurementWindow.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const editMode = !!store.recordId
+
+  const tabs = [{ label: labels.measurement }, { label: labels.measurementUnit, disabled: !editMode }]
+
+  return (
+    <>
+      <CustomTabs tabs={tabs} activeTab={activeTab} setActiveTab={setActiveTab} />
+
+      <CustomTabPanel index={0} value={activeTab}>
+        <MeasurementForm labels={labels} setStore={setStore} store={store} editMode={editMode} maxAccess={maxAccess} />
```
Comment (FatimaSalami) [inline src/pages/iv-measurements/Windows/MeasurementWindow.js:22]: it is better not to pass editMode.. instead define it in each form alone

File: src/pages/iv-measurements/Forms/MeasurementForm.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+export default function MeasurementForm({ labels, maxAccess, setStore, store, editMode }) {
+  const { recordId } = store
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: InventoryRepository.Measurement.page
+  })
+
+  const { formik } = useForm({
```
Comment (FatimaSalami) [inline src/pages/iv-measurements/Forms/MeasurementForm.js:25]: we should pass max access to formik

File: src/pages/iv-measurements/Forms/MeasurementUnit.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+  const openForm = id => { 
+    stack({
+      Component: MeasurementUnitForm,
+      props: {
+        labels,
+        maxAccess: maxAccess,
+        recordId: id,
+        msId: recordId,
+        editMode: editMode,
```
Comment (FatimaSalami) [inline src/pages/iv-measurements/Forms/MeasurementUnit.js:67]: i guess it is better to define editMode in form

File: src/pages/iv-measurements/Forms/MeasurementUnit.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        getMeasurementUnitGridData
+      },
+      width: 450,
+      height: 330,
+      title: labels.measurementUnit
+    })
+  }
+
+  useEffect(() => {
+    recordId && getMeasurementUnitGridData(recordId)
```
Comment (FatimaSalami) [inline src/pages/iv-measurements/Forms/MeasurementUnit.js:77]: better to use if recordid

**Pull request merged**